pkg/network/port_test.go

```diff
@@ -120,17 +121,15 @@ func TestReadInitialUDPState(t *testing.T) {
 	err := exec.Command("testdata/setup_netns.sh", nsName).Run()
 	require.NoError(t, err, "setup_netns.sh failed")
 
-	l, err := net.ListenUDP("udp", &net.UDPAddr{})
-	require.NoError(t, err)
-	defer func() { _ = l.Close() }()
+	l := nettestutil.StartServerUDP(t, net.ParseIP("0.0.0.0"), 5432)
```

Comment:
can you pass in 0 for the port here and let StartServerUDP select a free port? Whenever possible tests should 
let the OS decide on a port to use instead of hard coding.  

If another test is using 5432, and didn't clean up correctly, this test would fail for hard-to-see reasons.

Same comment below.

Reply:
I tried to use 0 for ports, but I get an assertion error:
```
panic: Assertion failed

goroutine 82 [running]:
github.com/stretchr/testify/assert.(*CollectT).FailNow(0xffff65e4f298?)
	/home/vagrant/go/pkg/mod/github.com/stretchr/testify@v1.8.4/assert/assertions.go:1874 +0x2c
github.com/stretchr/testify/require.NoError({0x1575328, 0x40006e93e0}, {0x1571ba0, 0x40003a1d60}, {0x0, 0x0, 0x0})
	/home/vagrant/go/pkg/mod/github.com/stretchr/testify@v1.8.4/require/require.go:1360 +0xac
github.com/DataDog/datadog-agent/pkg/network/testutil.PingUDP({0x1575328?, 0x40006e93e0}, {0x40001d4470, 0x10, 0x10}, 0x0)
	/git/datadog-agent/pkg/network/testutil/ping.go:70 +0x1bc
github.com/DataDog/datadog-agent/pkg/network/testutil.StartServerUDP.func2(0x0?)
	/git/datadog-agent/pkg/network/testutil/server.go:134 +0x38
github.com/stretchr/testify/assert.EventuallyWithT.func1()
	/home/vagrant/go/pkg/mod/github.com/stretchr/testify@v1.8.4/assert/assertions.go:1933 +0x34
created by github.com/stretchr/testify/assert.EventuallyWithT
	/home/vagrant/go/pkg/mod/github.com/stretchr/testify@v1.8.4/assert/assertions.go:1932 +0x20c
FAIL	github.com/DataDog/datadog-agent/pkg/network	0.043s
FAIL
```

Reply:
And if it does go through it returns false in port mappings:
```
    port_test.go:153: PortMapping(testRootNs, p) returned false for port 0
    port_test.go:153: PortMapping(testRootNs, p) returned false for port 0
 ```

Reply:
You'll need to re-work PingUDP and StartServerUDP slightly to get it to work, but after running locally it seems that the error is recoverable. PingUDP is using require. It shouldn't be because PingUDP is called inside require.EVentuallyT